File: src/main/java/com/att/training/katas/primes/PrimeFactors.java
package com.att.training.katas.primes;

import java.util.ArrayList;
import java.util.List;

class PrimeFactors {

    private PrimeFactors() {
        // No instances allowed
    }

    static List<Integer> of(int num) {
        var factors = new ArrayList<Integer>();

        for (int divisor = 2; num > 1; divisor++) {
            for (; num % divisor == 0; num /= divisor) {
                factors.add(divisor);
            }
        }

        return factors;
    }
}

File: src/test/kotlin/primes/PrimeFactors.kt
package com.att.training.katas.kt.primes

object PrimeFactors {
    fun of(input: Int): List<Int> {
        val factors = mutableListOf<Int>()
        var dividend = input
        var divisor = 2

        while (dividend > 1) {
            while (dividend % divisor == 0) {
                factors.add(divisor)
                dividend /= divisor
            }
            divisor++
        }

        return factors
    }
}
